### ISSUE TITLE
Some items are invisible or render incorrectly in the loot table preview 

### ISSUE THREAD

**Issue by Apollounknowndev** [misode/misode.github.io#301]
Invisible items (incomplete list):
- beds
- shields
- skulls and heads

Missing item colors:
- tipped arrows
- potions
- spawn eggs
- leather armor
- firework star
- mangrove leaves
- filled map

![image](https://user-images.githubusercontent.com/102649729/200645161-4bbd5f6d-1f26-4f38-837b-6134931e0fd3.png)

**Comment by ApolloDatapacks** [comment]
Skulls do not render either.

**Comment by misode** [comment]
This affects all block entities (using special entity renderers)

**Comment by catter1** [comment]
Not sure if more examples are needed, but found it also applies to potions. Uses this loot table for example:
```
{
  "type": "minecraft:entity",
  "pools": [
    {
      "rolls": 1,
      "entries": [
        {
          "type": "minecraft:item",
          "name": "minecraft:potion",
          "functions": [
            {
              "function": "minecraft:set_potion",
              "id": "minecraft:fire_resistance"
            }
          ]
        }
      ]
    }
  ]
}
```
![image](https://user-images.githubusercontent.com/63272345/203055538-ff1c5006-05f2-4909-aa70-e7dc29695d27.png)

**Comment by misode** [comment]
Part of this issue has been implemented in b165d551e4cd8f4db0051e04c412a9f8a44307ce.
A loot table demonstrating this: https://misode.github.io/loot-table/?share=oQTVAdG5qr
<img width="362" alt="chrome_d3bfDwChUQ" src="https://user-images.githubusercontent.com/17352009/203469258-da247a81-0558-438f-a797-43951cbf75a6.png">
